### PR TITLE
[feature](decimal) support creating table columns with type decimal256

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run p0

**Comment by jacktengg** [comment]
run pipelinex_p0

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run feut

**Comment by jacktengg** [comment]
run feut

**Comment by BiteTheDDDDt** [comment]
maybe we can add `using Int256 = wide::Int256`?

**Comment by jacktengg** [comment]
> maybe we can add `using Int256 = wide::Int256`?

We already had `struct UInt256` in uint128.h, in wide_integer.h also had `Int256` and `UInt256` which are all in the `wide` namespace, `using Int256 = wide::Int256` can be confusing, better to use `wide::Int256` when doing calculations.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java [hunk 1]
```diff
@@ -2155,7 +2156,8 @@ public Type getCompatibleType(Type lastCompatibleType, Expr lastCompatibleExpr,
         if (lastCompatibleType == null) {
             newCompatibleType = expr.getType();
         } else {
-            newCompatibleType = Type.getAssignmentCompatibleType(lastCompatibleType, expr.getType(), false);
+            newCompatibleType = Type.getAssignmentCompatibleType(
```
[the remark this answers is not in the document]
Comment (jacktengg) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java:2159]: Type.java is in fe/fe-common and it cannot import classes in fe/fe-core.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ArithmeticExpr.java [hunk 2]
```diff
@@ -111,7 +111,7 @@ public static void initBuiltins(FunctionSet functionSet) {
             for (int j = 0; j < Type.getNumericTypes().size(); j++) {
                 Type t2 = Type.getNumericTypes().get(j);
 
-                Type retType = Type.getNextNumType(Type.getAssignmentCompatibleType(t1, t2, false));
+                Type retType = Type.getNextNumType(Type.getAssignmentCompatibleType(t1, t2, false, false));
```
[the remark this answers is not in the document]
Comment (jacktengg) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ArithmeticExpr.java:114]: It's in the old planner and doing init, pass `false` to keep the original logic.

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**